### PR TITLE
Allow plugins to store multiple credentials on creation

### PR DESCRIPTION
**Pull request by tcsc**
The Okta service needs to store more than one credential associated with
the plugin, so this patch alters the Plugin request create message type
to allow this.

### CODE REVIEW

**Review by Tener — APPROVED** [review]

**Comment by Tener** [comment]
I assume we don't want to break backwards compat by deprecating the old field?

**Comment by tcsc** [comment]
> I assume we don't want to break backwards compat by deprecating the old field?

Correct. You can see the backwards-compatibility hoop-jumping in the corresponding PR in `teleport.e`: gravitational/teleport.e#2930

File: api/proto/teleport/plugins/v1/plugin_service.proto [hunk 1]
```diff
@@ -43,9 +43,18 @@ message CreatePluginRequest {
...
+  // StaticCredentials are an optional static credential to supply to the plugin.
   types.PluginStaticCredentialsV1 static_credentials = 3;
+
+  // StaticCredentials are an optional collection of static credentials
+  // to supply to the plugin. Ignored if `StaticCredentials` is set.
+  repeated types.PluginStaticCredentialsV1 static_credentials_list = 4;
+
+  // CredentialLabels are a collection of labels used to identify the
+  // credentials secified in the StaticCredentialsList. Ignored if
+  // StaticCredentials is used
```
Comment (codingllama) [inline api/proto/teleport/plugins/v1/plugin_service.proto:56]:
```suggestion
  // StaticCredentials are used (either static_credentials or static_credentials_list).
```

I suspect?

File: api/proto/teleport/plugins/v1/plugin_service.proto [hunk 2]
```diff
@@ -43,9 +43,18 @@ message CreatePluginRequest {
...
+  map<string, string> credential_labels = 5;
```
Comment (codingllama) [inline api/proto/teleport/plugins/v1/plugin_service.proto:57]: I'm not a fan of proto maps, so I'd much rather do `repeated Label credential_labels = 5;`, where Label is something like this:

```proto
message Label {
  string key = 1;
  string value = 2;
}
```

The API future proofing of maps is rather bad.

That said, we have plenty of labels defined like this, so I'll leave this up to you.

**Review by codingllama — APPROVED** [review]

File: api/proto/teleport/plugins/v1/plugin_service.proto (hunk 2 above)
Comment (tcsc) [reply api/proto/teleport/plugins/v1/plugin_service.proto:57]: I definitely see your argument re: `repeated Label`, but given that pretty much everything else encodes labels as a `map<string,string>` I'm going to go with consistency over "correctness" (if that's even the right word?) in this case.

**Pull request merged**